csharp/tests/Integration/IntegrationTestBase.cs

```diff
@@ -0,0 +1,135 @@
+﻿/**
+ * Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+ */
+
+using System.Diagnostics;
+
+// Note: All IT should be in the same namespace
+namespace tests.Integration;
+
+[SetUpFixture]
+public class IntegrationTestBase
+{
+    [OneTimeSetUp]
+    public void SetUp()
+    {
+        // Stop all if weren't stopped on previous test run
+        StopRedis(false);
+
+        // Delete dirs if stop failed due to https://github.com/aws/glide-for-redis/issues/849
+        Directory.Delete(Path.Combine(scriptDir, "clusters"), true);
+
+        // Start cluster
+        CLUSTER_PORTS = StartRedis(true);
+        // Start standalone
+        STANDALONE_PORTS = StartRedis(false);
+
+        // Get redis version
+        REDIS_VERSION = GetRedisVersion();
+
+        TestContext.Progress.WriteLine($"Cluster ports = {string.Join(',', CLUSTER_PORTS)}");
+        TestContext.Progress.WriteLine($"Standalone ports = {string.Join(',', STANDALONE_PORTS)}");
+        TestContext.Progress.WriteLine($"Redis version = {REDIS_VERSION}");
+    }
+
+    [OneTimeTearDown]
+    public void TearDown()
+    {
+        // Stop all
+        StopRedis(true);
+    }
+
+    public static List<uint> STANDALONE_PORTS { get; private set; } = new();
+    public static List<uint> CLUSTER_PORTS { get; private set; } = new();
+    public static Version REDIS_VERSION { get; private set; } = new();
+
+    private readonly string scriptDir;
+
+    // Nunit requires a public default constructor. These variables would be set in SetUp method.
+    public IntegrationTestBase()
+    {
+        STANDALONE_PORTS = new();
```

Comment:
why re-init?

Reply:
I'm following there a bad coding practice - using static members as class ones, because this class is a singletone-like.
Unfortunately, I can't make it as a singletone, because nunit requires a public constructor.

Reply:
Reworked that, should be better now